tests/security/plugin_trusted_path.cpp

```
@@ -0,0 +1,99 @@
+#include <iostream>
+#include <string>
+#include <filesystem>
+#include <set>
+#include <functional>
+
+// Reproduce isTrustedPath from PluginScanner.cpp
+bool isTrustedPath(const std::filesystem::path& path) {
+    std::string p = path.lexically_normal().string();
+
+    // Linux: system-wide paths are trusted
+    if (p.find("/usr/lib/vst3") == 0 ||
```

Cppcheck (2.20.0)

[performance] Inefficient usage of string

(stlIfStrFind)